### PR TITLE
added selectors2-pip library

### PR DESCRIPTION
**Pull request by ejalaa12**
Added the selectors2 pip library which is a port of the python3 selectors library to python2.

See the project on [Pypi](https://pypi.org/project/selectors2/)
> Backported, durable, and portable selectors designed to replace the standard library selectors module.

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -5543,6 +5543,22 @@ rpy2:
   debian: [python-rpy2]
   gentoo: [=dev-python/rpy-2*]
   ubuntu: [python-rpy2]
+selectors2-pip:
```
Comment (tfoote) [inline rosdep/python.yaml]: Please use the `python-` prefix. We generally want to stick with the debian style name for consistency.

**Review by tfoote — CHANGES_REQUESTED** [review]

**Comment by nuclearsandwich** [comment]
@ejalaa12 friendly ping. Are you able to address tfoote's feedback or may we close the PR?

**Comment by ejalaa12** [comment]
@nuclearsandwich , sorry, it's been a busy month. I'll do the change.
However, @tfoote maybe I might have misunderstood the [Contributing.md](https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md) as it says:

> If a package is only available via pip use the -pip extension.

That's why I didn't add the `python-` prefix. Should I then rename the entry to `python-selectors2` or `python-selectors2-pip` ?

**Comment by clalancette** [comment]
> > 
> 
> That's why I didn't add the `python-` prefix. Should I then rename the entry to `python-selectors2` or `python-selectors2-pip` ?

What we generally aim to do is to have the rosdep keyname match the debian/ubuntu package name (there are *lots* of exceptions to this in the database, but we can't go back and fix them now without breaking everyone).

With things that are only available in pip, it gets a little more unclear.  In this case, the debian package would *probably* be `python-selectors2` (given that it is a Python library), so our preferred key name would be `python-selectors2-pip`.

**Comment by ejalaa12** [comment]
@clalancette I just pushed those modifications. I get it now, it is in anticipation of the debian package being published. But then once it is published we should remove the `-pip` prefix, no ?

**Comment by clalancette** [comment]
> @clalancette I just pushed those modifications. I get it now, it is in anticipation of the debian package being published. But then once it is published we should remove the `-pip` prefix, no ?

Sort-of.  We'll add a second entry without the `-pip` prefix, to avoid breaking packages that are already depending on the pip one.

**Review by clalancette — APPROVED** [review]

**Review by tfoote — APPROVED** [review]

**Pull request merged**